src/python/pants/backend/python/lint/black/rules_integration_test.py

```diff
@@ -202,3 +202,60 @@ class Foo:
     assert "1 file left unchanged" in fmt_result.stderr
     assert fmt_result.output == get_digest(rule_runner, [py38_sources])
     assert fmt_result.did_change is False
+
+
+def test_stub_files(rule_runner: RuleRunner) -> None:
+    good_stub = replace(GOOD_SOURCE, path="good.pyi")
+    bad_stub = replace(BAD_SOURCE, path="bad.pyi")
+    fixed_bad_stub = replace(FIXED_BAD_SOURCE, path="bad.pyi")
+    needs_config_stub = replace(NEEDS_CONFIG_SOURCE, path="needs_config.pyi")
+
+    # Passing
+    good_files = [GOOD_SOURCE, good_stub]
+    target = make_target(rule_runner, good_files)
+    lint_results, fmt_result = run_black(rule_runner, [target])
+    assert len(lint_results) == 1 and lint_results[0].exit_code == 0
+    assert (
+        "2 files would be left unchanged" in lint_results[0].stderr
+        and "2 files left unchanged" in fmt_result.stderr
+    )
+    assert fmt_result.output == get_digest(rule_runner, good_files)
+    assert not fmt_result.did_change
+
+    # Failing
+    target = make_target(rule_runner, [BAD_SOURCE, bad_stub], name="failing_target")
+    lint_results, fmt_result = run_black(rule_runner, [target])
+    assert len(lint_results) == 1 and lint_results[0].exit_code == 1
+    assert (
+        "2 files would be reformatted" in lint_results[0].stderr
+        and "2 files reformatted" in fmt_result.stderr
+    )
+    fixed_bad_files = [FIXED_BAD_SOURCE, fixed_bad_stub]
+    assert fmt_result.output == get_digest(rule_runner, [*fixed_bad_files, *good_files])
+    assert fmt_result.did_change
+
+    # Mixed targets + Multiple targets + Needs config
```

Comment:
I think it's likely best to leave off this edge case. At this point, it's on Black whether it handles config files properly for `.pyi` files or not, imo, as we've already verified that a) Pants correctly loads config files and b) Black can handle `.pyi` in general. 

What's the downside to this extra test? It slightly slows down our CI (slow CI was a big problem with the v1 engine), and it's more code to maintain. 

But, there is some upside that we're more comprehensive. It's a tradeoff. I suspect that here it's better to remove, although there's an argument to keep this.

Reply:
There's an argument we don't even need both a `passing` and `failing` for stubs, but I think that's more acceptable to have both because it's very core functionality, rather than an edge case like if a config file was used.

Reply:
Could we add it to a separate test case? Would that make it easier to diagnose and remove later if it causes a CI slowdown?

But separately, I totally buy this:
> At this point, it's on Black whether it handles config files properly for .pyi files or not, imo, as we've already verified that a) Pants correctly loads config files and b) Black can handle .pyi in general.

So I would be fine leaving it out.

Reply:
Agreed. I am happy to remove the `failing` part too.